### PR TITLE
fix story menu breaking when remix disabled

### PR DESCRIPTION
**Pull request by youbitchoc**
also clean up a bunch of other stuff

### CODE REVIEW

File: Story/StoryMenuHelpers.cs
[the remark this answers is not in the document]
Comment (youbitchoc) [reply Story/StoryMenuHelpers.cs:1]: > It’s possible you are updating the client page index elsewhere

yup, see SetCurrentlySelectedOfSeries

File: Story/StoryOnlineMenu.cs [hunk 1]
```diff
@@ -134,90 +116,168 @@ public override void ShutDownProcess()
...
+
+        public void SetCurrentlySelectedOfSeries(string series, int to)
+        {
+            switch (series)
             {
-                var playerbtn = playerButtons[i];
-                playerbtn.RemoveSprites();
-                this.pages[0].RemoveSubObject(playerbtn);
+                case "scugButtons":
+                    slugcatPageIndex = to;
```
Comment (6fears7) [inline Story/StoryOnlineMenu.cs:157]: I was expecting to see RemoveExcessStoryObjects called here again for when the page update triggers a new slugcat, but I only see it called when the client first joins the lobby. Can you verify?

Comment (youbitchoc) [reply Story/StoryOnlineMenu.cs:157]: I changed RemoveExcessStoryObjects to remove for all pages

Comment (6fears7) [reply Story/StoryOnlineMenu.cs:157]: Awesome

**Pull request merged**